### PR TITLE
Cleaning up upload container on failed sync validation

### PR DESCRIPTION
**Pull request by agr**
Addresses https://github.com/NuGet/NuGetGallery/issues/6570

Moved package contents processing (opening package, validating metadata, getting the license data) into separate method `ValidateAndProcessPackageContents`. Added code that if that method returns error the package is deleted from the upload container, preventing the situation when Gallery tries to resume package upload for a package that was deemed invalid.

### CODE REVIEW

File: src/NuGetGallery/Controllers/PackagesController.cs [hunk 1]
```diff
@@ -544,21 +544,71 @@ private async Task<JsonResult> UploadPackageInternal(PackageArchiveReader packag
             PackageRegistration existingPackageRegistration,
             bool isSymbolsPackageUpload,
             bool hasExistingSymbolsPackageAvailable)
+        {
+            var packageContentData = await ValidateAndProcessPackageContents(currentUser, isSymbolsPackageUpload);
+
```
Comment (loic-sharma) [inline src/NuGetGallery/Controllers/PackagesController.cs]: Should we delete the upload file if `ValidateAndProcessPackageContents` throws?

Comment (agr) [reply src/NuGetGallery/Controllers/PackagesController.cs]: There is a catch-all in the top level entry function:
https://github.com/NuGet/NuGetGallery/blob/189ac93b7b408e5e10039c19dcf584e36b0ec3e1/src/NuGetGallery/Controllers/PackagesController.cs#L364-L367
That shows generic "Failed to read the package file" message.

Comment (agr) [reply src/NuGetGallery/Controllers/PackagesController.cs]: Well, never mind, it does not delete the file...

File: src/NuGetGallery/Controllers/PackagesController.cs [hunk 2]
```diff
@@ -545,7 +545,17 @@ private async Task<JsonResult> UploadPackageInternal(PackageArchiveReader packag
             bool isSymbolsPackageUpload,
             bool hasExistingSymbolsPackageAvailable)
         {
-            var packageContentData = await ValidateAndProcessPackageContents(currentUser, isSymbolsPackageUpload);
+            PackageContentData packageContentData = null;
+            try
+            {
+                packageContentData = await ValidateAndProcessPackageContents(currentUser, isSymbolsPackageUpload);
+
```
Comment (loic-sharma) [inline src/NuGetGallery/Controllers/PackagesController.cs]: Please remove line 552

**Pull request merged**